### DISCUSSION TITLE
White Screen Issue During Deployment to a Specific Domain via Vercel

### DISCUSSION THREAD

**Discussion by nrjshka** [vercel/community#2644] [Help] [answered]
### Summary

We're encountering an issue where deploying to router.via.exchange results in a white screen for users, even though the same bundle works perfectly when redirected to other domains (we have a temporary redirect set up to 0xvia.com). This problem started between Thursday and Friday, and we initially thought it might be related to the DNS issues Vercel reported on their status page. We attempted to redirect back to router.via.exchange, but this did not solve the issue. We have screenshots detailing the problem and are seeking your assistance.

### Example

To reproduce the issue, you can visit the site at router.via.exchange. You should see a white screen. However, if you visit 0xvia.com, which is temporarily redirecting to the same bundle, it works as expected.

https://app.birdeatsbug.com/sessions/aaCHtu3xsMju2NSqXGHT8iC5iTlnm8Amf3OD3-od5rRT

### Steps to Reproduce

1. Deploy to router.via.exchange via Vercel.
2. Visit the site and observe the white screen issue.
3. Redirect to another domain (e.g., 0xvia.com) with the same bundle.
4. Visit the redirected site and observe that it works correctly.
Please find attached screenshots for further clarification. We hope to resolve this as soon as possible and appreciate your help!

**Comment by smaeda-ks** [comment]
Hi @nrjshka, unfortunately, it's hard to investigate unless the issue can actually be seen by us. Do you have any test URLs that I can see the white screen issue in live? Or, could you please provide a minimal reproducible example repo?
https://vercel.com/guides/creating-a-minimal-reproducible-example

**Reply by nrjshka** [reply]
Try checking https://router.via.exchange/

**Comment by smaeda-ks** [comment, marked as answer]
@nrjshka 
Seems like you had some bad deployment earlier and some of your static JS assets may be broken and cached.
![Screenshot 2023-05-29 at 22 52 48](https://github.com/vercel/community/assets/11495867/afd85803-1049-4ab8-a62b-7f4caa0ea842)

Could you trigger another deployment that would change the file hash to get rid of those old caches?

**Reply by smaeda-ks** [reply]
also, make sure to purge Cloudflare caches

**Reply by nrjshka** [reply]
But try checking 0xvia.com, it's the same code and it's running okay

**Reply by smaeda-ks** [reply]
cache is created per hostname, different hostname can get different cache, same for Cloudflare cache

**Reply by smaeda-ks** [reply]
in fact, if you update your local hosts file, you can access https://router.via.exchange/ just fine.

```
$ cat /private/etc/hosts

76.76.21.21 router.via.exchange
```

so please make sure to purge Cludflare cache as well

**Reply by nrjshka** [reply]
I believe it may not be related to the cache issue, as you could try opening this website on different profiles of the **same browser**, or perhaps try opening it in incognito mode. There seems to be an unusual occurrence and this error appears intermittently. If it would be helpful, I would be more than happy to arrange a call and provide a demonstration from my working computer for you to compare, if that would assist you.

**Reply by smaeda-ks** [reply]
@nrjshka you need to check your Cloudflare configuration first. As mentioned, accessing directly to your Vercel deployment by overriding the local hosts file just works fine, which indicates the issue is something has to do with your Cloudflare layer. Please perform necessary troubleshooting steps on your end before asking further. Here is more supporting evidence:

```
# via Cloudflare
$ curl -s https://router.via.exchange/assets/index-1d1799ea.js | md5sum
7001073025ce3b2a7c326164671e77f7  -

# Vercel direct access
$ curl -s https://router.via.exchange/assets/index-1d1799ea.js  --connect-to router.via.exchange:443:cname.vercel-dns.com | md5sum
50a186729098e166297cd09007601160  -

$ curl -s https://0xvia.com/assets/index-1d1799ea.js  --connect-to 0xvia.com:443:cname.vercel-dns.com | md5sum
50a186729098e166297cd09007601160  -
```

you can  clearly see Cloudflare version is serving different content (different hash).

edit:
I made some typo in the command output above and updated them. But the symptom remains the same.
